compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -1888,6 +1897,29 @@ public BSymbol getArithmeticOpsForTypeSets(OperatorKind opKind, BType lhsType, B
         return symTable.notFoundSymbol;
     }
 
+    public BSymbol getUnaryArithmeticOpsForTypeSets(OperatorKind opKind, BType type) {
+        boolean validIntTypesExists;
+        switch (opKind) {
+            case ADD:
+            case SUB:
+                validIntTypesExists = types.validNumericTypeExists(type);
+                break;
+            default:
+                return symTable.notFoundSymbol;
+        }
+        if (validIntTypesExists) {
```

Comment:
The following still doesn't seem to work?

```ballerina
type A 1456|23456;

type B byte|A;

public function main() {
    B v = 0;
    int _ = -v;
}
```

Reply:
Could be an issue with other operations that use ` types.validNumericTypeExists` also.

Reply:
Yeah, there is an issue with the logic of `types.validNumericTypeExists` since it is being used for resolving binary operators. Following also not working. 
  ```
    B v = 0;
    B w = 0;
    int _ = w+v;
```